sway-lib-std/src/storage/storage_vec.sw

```diff
@@ -865,14 +887,44 @@ impl<V> StorageKey<StorageVec<V>> {
             0 => Vec::new(),
             len => {
                 // Get the number of storage slots needed based on the size.
-                let bytes = len * __size_of::<V>();
+                let size_V_bytes = __size_of::<V>();
+                let bytes = if size_V_bytes < 8 {
+                    // Len * size_of_word
+                    len * 8
+                } else {
+                    len * size_V_bytes
+                };
```

Comment:
I don't think it's a requirement but It may be a good opportunity to add max and min to core similar to Rust's: https://doc.rust-lang.org/std/cmp/index.html

Reply:
https://github.com/FuelLabs/sway/issues/6169